### PR TITLE
store-gateway: remove cortex_bucket_store_blocks_loaded_by_duration

### PR DESCRIPTION
**Pull request by dimitarvdimitrov**
In the PR which added this metric #6074, the motivation was to help detect compactors which are falling behind. There is another metric `cortex_bucket_store_series_blocks_queried` added by @jhalterman in #7112 which serves better the purpose of detecting uncompacted blocks. The reason is that it has an explicit label for uncompacted blocks (`source`), and it also is more sensitive to recent blocks (assuming those are queried much more often than old uncompacted blocks).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -37,6 +37,7 @@
 * [CHANGE] Query-frontend: the default value of the CLI flag `-query-frontend.max-cache-freshness` (and its respective YAML configuration parameter) has been changed from `1m` to `10m`. #7161
 * [CHANGE] Distributor: default the optimization `-distributor.write-requests-buffer-pooling-enabled` to `true`. #7165
 * [CHANGE] Tracing: Move query information to span attributes instead of span logs. #7046
+* [CHANGE] Store-gateway: remove `cortex_bucket_store_blocks_loaded_by_duration`. You can use `cortex_bucket_store_series_blocks_queried` instead. #7309
```
Comment (56quarters) [inline CHANGELOG.md:40]:
```suggestion
* [CHANGE] Store-gateway: remove `cortex_bucket_store_blocks_loaded_by_duration`. `cortex_bucket_store_series_blocks_queried` is better suited for detecting when compactors are not able to keep up with the number of blocks to compact. #7309
```

**Review by 56quarters — APPROVED** [review]
LGTM

File: pkg/storegateway/bucket_stores.go [hunk 2]
```diff
@@ -549,37 +542,29 @@ func (u *BucketStores) closeBucketStoreAndDeleteLocalFilesForExcludedTenants(inc
 
 // countBlocksLoaded returns the total number of blocks loaded and the number of blocks
 // loaded bucketed by the provided block durations, summed for all users.
```
Comment (pr00se) [inline pkg/storegateway/bucket_stores.go:544]: Is this comment still correct?

Comment (dimitarvdimitrov) [reply pkg/storegateway/bucket_stores.go:544]: oooh, nice catch. It isn't

**Review by pracucci — APPROVED** [review]

**Pull request merged**